apis/vpcresources/v1alpha1/cninode_types.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FeatureName is a type of feature name supported by AWS VPC CNI. It can be Security Group for Pods, custom networking, or others
+type FeatureName string
+
+const (
+	SecurityGroupsForPods FeatureName = "SecurityGroupsForPods"
+	CustomNetworking      FeatureName = "CustomNetworking"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+// CNINodeSpec defines the desired state of CNINode
+type CNINodeSpec struct {
+	Features []FeatureName `json:"features,omitempty"`
```

Comment:
shall we have list of features or  something like
```
spec:
 securityGroupsForPods:
    enabled: true
 customNetworking:
    enabled: false
```
the benefit i can see is we can optionally have more configurations for each feature when necessary.

Reply:
yes, we discussed this as well. For this version as alpha1, we intend to onboard the two features and can reference them as feature flags then the boolean value is not a must. But I agree we can definitely add support on values when we onboard more features or combined configurations which need track their status.

Reply:
Synced offline, we can accept the current design for now and keep the door open.